Source/WTF/wtf/text/WTFString.h

```diff
@@ -491,11 +491,11 @@ inline String String::substring(unsigned position, unsigned length) const
 }
 
 template<typename Func>
-inline Expected<std::invoke_result_t<Func, std::span<const char>>, UTF8ConversionError> String::tryGetUTF8(const Func& function, ConversionMode mode) const
+inline Expected<std::invoke_result_t<Func, std::span<const char8_t>>, UTF8ConversionError> String::tryGetUTF8(const Func& function, ConversionMode mode) const
 {
     if (!m_impl) {
-        constexpr const char* emptyString = "";
-        return function(std::span(emptyString, emptyString));
+        constexpr std::array<char8_t, 1> empty { '\0' };
+        return function(std::span { empty.data(), empty.data() });
```

Comment:
I think we may not need the zero byte at the end. I suspect it is enough to have a non-null pointer inside the span so the CString constructor decides to constructor an empty string instead of a null string. Hopefully that constructor doesn't rely on the null terminator given than it's not included in the span. I'll give it a try.